### PR TITLE
improve(sdk): calculate fees using the same logic as the bots

### PR DESCRIPTION
**Pull request by amateima**
**Motivation**

https://app.shortcut.com/uma-project/story/3420/make-sure-the-fees-in-ui-matches-with-how-the-bots-are-calculating-fees

**Summary**

Match the fees with the ones calculated by the bots.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/sdk/src/across/gasFeeCalculator.ts [hunk 1]
```diff
@@ -23,7 +25,20 @@ export default async (
   gas: number,
   tokenAddress?: string
 ): Promise<ReturnType> => {
-  const gasPrice = await ethersProvider.getGasPrice();
+  const latestBlockNumber = await (ethersProvider as ethers.providers.JsonRpcProvider).getBlockNumber();
+  const { baseFeePerGas } = await (ethersProvider as ethers.providers.JsonRpcProvider).getBlock(latestBlockNumber);
+  let gasPrice: ethers.BigNumber;
+
+  if (baseFeePerGas) {
```
Comment (chrismaree) [inline packages/sdk/src/across/gasFeeCalculator.ts:32]: this is perfect! good work on creating the `etherchain` client as well.

File: packages/sdk/src/across/gasFeeCalculator.ts [hunk 2]
```diff
@@ -23,7 +25,20 @@ export default async (
...
+    const priorityFeePerGas = (await new Etherchain().getGasPrice()).fastest;
+    // transform priority fee from gwei (eg 4.1) to wei
+    const priorityFeePerGasWei = Math.round(priorityFeePerGas * 10 ** 9);
+    gasPrice = baseFeePerGas.add(priorityFeePerGasWei);
+  } else {
+    // fallback in case baseFeePerGas is undefined / null
+    gasPrice = await ethersProvider.getGasPrice();
```
Comment (chrismaree) [inline packages/sdk/src/across/gasFeeCalculator.ts:39]: 👌

**Review by chrismaree — APPROVED** [review]
this is great! good work and thanks for getting this out so fast.

File: packages/sdk/src/across/gasFeeCalculator.ts [hunk 3]
```diff
@@ -23,7 +25,20 @@ export default async (
   gas: number,
   tokenAddress?: string
 ): Promise<ReturnType> => {
-  const gasPrice = await ethersProvider.getGasPrice();
+  const latestBlockNumber = await (ethersProvider as ethers.providers.JsonRpcProvider).getBlockNumber();
+  const { baseFeePerGas } = await (ethersProvider as ethers.providers.JsonRpcProvider).getBlock(latestBlockNumber);
```
Comment (daywiss) [inline packages/sdk/src/across/gasFeeCalculator.ts:29]: You can usually do `getBlock("latest")`, might save a call

File: packages/sdk/src/clients/etherchain/client.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
+import { get } from "lodash";
```
Comment (daywiss) [inline packages/sdk/src/clients/etherchain/client.ts:2]: I think better for bundling to use 'from lodash/get'

File: packages/sdk/src/clients/etherchain/client.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  fast: number;
+  fastest: number;
+  currentBaseFee: number;
+  recommendedBaseFee: number;
+};
+
+export class Etherchain {
+  private url: string;
+
+  constructor(url = "https://www.etherchain.org/api") {
```
Comment (daywiss) [inline packages/sdk/src/clients/etherchain/client.ts:16]: You can shortcut defining the private var by saying 'private url' here and remove the declaration of the var above

**Review by daywiss — COMMENTED** [review]
Very nice, just a couple nits

File: packages/sdk/src/across/gasFeeCalculator.ts (hunk 3 above)
Comment (amateima) [reply packages/sdk/src/across/gasFeeCalculator.ts]: thx

File: packages/sdk/src/clients/etherchain/client.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+type GasPrice = {
+  safeLow: number;
+  standard: number;
+  fast: number;
+  fastest: number;
+  currentBaseFee: number;
+  recommendedBaseFee: number;
+};
+
+export class Etherchain {
```
Comment (mrice32) [inline packages/sdk/src/clients/etherchain/client.ts:13]: Are we planning to use this in the bots? Shouldn't the bots and FE use the same logic if we want them to behave the same?

**Review by mrice32 — COMMENTED** [review]
Just one high level comment!

File: packages/sdk/src/clients/etherchain/client.ts (hunk 6 above)
Comment (amateima) [reply packages/sdk/src/clients/etherchain/client.ts:13]: > Are we planning to use this in the bots?
I don't know how to answer to this

**Review by daywiss — APPROVED** [review]
nice work

**Pull request merged**